comp/core/workloadmeta/collectors/internal/process/process_collector.go

```diff
@@ -0,0 +1,237 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+// Package process implements the process collector for Workloadmeta.
+package process
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+	"github.com/benbjohnson/clock"
+	"go.uber.org/fx"
+
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	collectorID       = "process-collector"
+	componentName     = "workloadmeta-process"
+	cacheValidityNoRT = 2 * time.Second
+)
+
+type collector struct {
+	id                     string
+	store                  workloadmeta.Component
+	catalog                workloadmeta.AgentType
+	clock                  clock.Clock
+	processProbe           procutil.Probe
+	processEventsCh        chan *Event
+	lastCollectedProcesses map[int32]*procutil.Process
+}
+
+// Event is a message type used to communicate with the stream function asynchronously
+type Event struct {
+	Created []*workloadmeta.Process
+	Deleted []*workloadmeta.Process
+}
+
+func newProcessCollector(id string, store workloadmeta.Component, catalog workloadmeta.AgentType, clock clock.Clock, processProbe procutil.Probe, processEventsCh chan *Event, lastCollectedProcesses map[int32]*procutil.Process) collector {
```

Comment:
Suggestion, optional: `processEventsCh` and `lastCollectedProcesses` could be allocated inside this function instead of passed in. In my opinion it makes the function cleaner and you're allocating these objects within or immediately before the call anyway.